markdown_lab/network/async_client.py

```
@@ -0,0 +1,533 @@
+        
+        Args:
+            url: Target URL
+            use_cache: Whether to use cache for this request
+            **kwargs: Additional request arguments
+            
+        Returns:
+            Response content as string
+        """
+        # Check cache first
+        if use_cache and self.cache:
+            if cached_content := self.cache.get(url):
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)